aws/resource_aws_codebuild_project.go

```diff
@@ -548,6 +606,19 @@ func flattenAwsCodebuildProjectSource(source *codebuild.ProjectSource) []interfa
 	return l
 }
 
+func flattenAwsCodeBuildVpcConfig(vpcConfig *codebuild.VpcConfig) []interface{} {
+	if vpcConfig != nil {
+		values := map[string]interface{}{}
+
+		values["vpc_id"] = *vpcConfig.VpcId
+		values["subnets"] = schema.NewSet(schema.HashString, flattenStringList(vpcConfig.Subnets))
```

Comment:
Does this work with just `values["subnets"] = flattenStringList(vpcConfig.Subnets)`?

Reply:
```--- FAIL: TestAccAWSCodeBuildProject_basic (44.01s)
        testing.go:513: Step 0 error: Error applying: 1 error(s) occurred:

                * aws_codebuild_project.foo: 1 error(s) occurred:

                * aws_codebuild_project.foo: Invalid address to set: []string{"vpc_config", "0", "subnets"}
        testing.go:573: Error destroying resource! WARNING: Dangling resources
                may exist. The full state and error is shown below.

                Error: Error refreshing: 1 error(s) occurred:

                * aws_codebuild_project.foo: 1 error(s) occurred:

                * aws_codebuild_project.foo: aws_codebuild_project.foo: Invalid address to set: []string{"vpc_config", "0", "subnets"}

                State: <nil>
FAIL
```
I don't think so sadly :(

Reply:
It appears `d.Set` has the magic buried in there to make this work. I tried a few tricks to pull it out so it would work for subresources, but I couldn't quite make it work. I've stopped and decided that perhaps core should some features to set and get with subresources instead of my hack-ish attempts. If you think we should invest more here let me know, but right now I didn't want to block this with that.